analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs

```diff
@@ -28,7 +28,7 @@ public class RuleTypeTest
     {
         // Rules that have been deprecated and deleted.
         // When changing this please do not forget to notify the product teams (SQ, SC, SL).
-        private static readonly HashSet<string> DeletedRules = new() { "S1145", "S1697", "S4142", "S2758", "S2070", "S3693", "S4432", "S2278" };
+        private static readonly HashSet<string> DeletedRules = new() { "S1145", "S1697", "S2070", "S2255", "S2278", "S2758", "S3693", "S4142", "S4432"};
```

Comment:
What's the purpose of this field? If i delete it, nothing breaks.
So it doesn't make sense to add anything here => just delete the field, and keep the test. In case we'd delete rule, it would bite us.

Also do not forget to plan to notify SQ, SC and SL teams, as mentioned in the comment below.

Reply:
It probably lost its meaning during some of the RSPEC reworks in the past

Reply:
There's some logic there that uses this hashset.
I prefer to merge this PR as is and add a task to my calendar to check the actual usage of this hashset and remove it eventually in its own PR.

Reply:
That's what I did when asking for this. Just remove the field completely. If you clear all the elements, the UT still works => it doesn't have a purpose